src/adapter.rs

```diff
@@ -1163,7 +1163,21 @@ impl<'a> Adapter<'a> for RustdocAdapter<'a> {
                                             token.as_impl().expect("not an Impl token");
 
                                         if let Some(path) = &impl_token.trait_ {
-                                            if let Some(item) = item_index.get(&path.id) {
+                                            // When the implemented trait is from the same crate
+                                            // as its definition, the trait should be present
+                                            // in `item_index`. Otherwise, the
+                                            // `rustdoc_types::Trait` is not in this rustdoc. 
+                                            // The needed foreign traits for lints are hand-written 
+                                            // in `dummy_trait_items`.
```

Comment:
What is the intention behind `trustfall-rustdoc`? Is it a tool specifically for `cargo-semver-checks`? If yes, is it OK to explain that a fix was made only for the lints to work? If no, then I guess I shouldn't mention about the existence of lints?

Nevertheless, I don't really like this comment, but I can't find a better one. Can you phrase it better @obi1kenobi ?

Reply:
The comment changed, this conversation is no longer relevant.